src/orquestra/sdk/_ray/_client.py

```diff
@@ -112,6 +113,21 @@ def run_dag_async(
         ):
             ray.workflow.run_async(dag_node, workflow_id=workflow_id, metadata=metadata)
 
+            # Give some time to ray to actually submit workflow
+            submission_timeout_in_sec = 10
+            retry_time_in_sec = 0.1
+            retries = 0
+            while True:
+                try:
+                    self.get_workflow_status(workflow_id)
+                    break
+                except workflow_exceptions.WorkflowNotFoundError:
+                    if retries * retry_time_in_sec >= submission_timeout_in_sec:
+                        raise
+                    else:
+                        time.sleep(retry_time_in_sec)
+                        retries += 1
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Is it possible to add tests for these lines?

Reply:
Good point - ill see what I can do here